### PR TITLE
Pin cosmos rust proto version, and implement serde traits on definitions

### PR DESCRIPTION
**Pull request by cbrit**

### CODE REVIEW

File: prost_build/src/main.rs [hunk 1]
```diff
@@ -84,6 +84,8 @@ fn compile_protos(out_dir: &Path, tmp_dir: &Path) {
     let mut config = prost_build::Config::default();
     config.out_dir(tmp_dir);
     config
+        .type_attribute(".", "#[derive(serde::Deserialize, serde::Serialize)]")
+        .type_attribute(".", "#[serde(rename_all = \"snake_case\")]")
```
Comment (EricBolten) [inline prost_build/src/main.rs:88]: Is there a reason to do this renaming? It seems like the best option for compatibility would be maintaining the default naming scheme.

**Review by EricBolten — COMMENTED** [review]
A couple questions.

File: somm_proto/src/prost/cosmos_proto.rs [hunk 2]
```diff
@@ -1 +0,0 @@
-
```
[the remark this answers is not in the document]
Comment (cbrit) [reply somm_proto/src/prost/cosmos_proto.rs:1]: I guess so, not sure what it was but the build didn't regenerate it.

File: prost_build/src/main.rs (hunk 1 above)
Comment (cbrit) [reply prost_build/src/main.rs:88]: snake case has been the default in steward protos in the past, so I was trying to make it consistent.

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**